### PR TITLE
[6.x.x] tails of subsequences off by one

### PR DESCRIPTION
**Pull request by line-o**
### Description:

Backport of #4850 

### Reference:

#4830 

### Type of tests:

XQSuite tests

### CODE REVIEW

**Review by duncdrum — COMMENTED** [review]
CI is stuck at dependency checks

**Review by adamretter — CHANGES_REQUESTED** [review]
Same small issues as discussed here - https://github.com/eXist-db/exist/pull/4850#pullrequestreview-1376080745

**Review by adamretter — APPROVED** [review]
Thanks for the improvements. Now LGTM :-)

**Comment by adamretter** [comment]
@line-o Just needs 1 more approval from a core dev and then we can merge - @reinhapa as you reviewed and approved the non-backport would you be able to approve this one

**Comment by line-o** [comment]
I want to point out that @duncdrum _is_ a member of the core team which makes him a core-dev to me.

**Pull request merged**
